btc_execution/sources/interpreter.move

```diff
@@ -31,11 +41,15 @@ fun eval(ip: &mut Interpreter, r: ScriptReader): bool {
     ip.reader = r; // init new  reader
     while(!r.end_stream()) {
         let op = ip.reader.nextOpcode();
-
         if (op == OP_DUP) {
             ip.op_dup();
-            break
         };
+        if (op == OP_EQUAL) {
```

Comment:
use if - else. There is no need to do other condition check if we already hit one.